### ISSUE TITLE
Sharing XDR objects from a dependency 

### ISSUE THREAD

**Issue by mootz12** [stellar/js-stellar-base#617]
**Describe the bug**
When attempting to create a package that manages building the complicated XDR objects required for Soroban, I am unable to use separate but identical versions of `stellar-base` to handle these XDR objects. The use case is the consumer (a dApp or node script, for example) likely already relies on `soroban-client` and/or `stellar-sdk`.

For more context see the discussion in discord here: https://discord.com/channels/897514728459468821/1113560546705428600

**What version are you on?**
```
"stellar-base": "9.0.0-soroban.3"
```

**To Reproduce**
1. Go the MVE Repository and follow the reproduction steps https://github.com/mootz12/mve-xdr-dependency

**Expected behavior**
The XDR object created by the dependency is valid

**Additional context**
The current workarounds are to re-export `xdr` that the dependency uses, or to have the package return the xdr objects as strings and expect the consumer to build them from the string.

**Comment by silence48** [comment]
can using yarn resolutions help?

**Comment by Shaptic** [comment]
@mootz12 is this still true?

**Comment by mootz12** [comment]
Yeah, I updated the `stellar-base` dependency to `10.0.0-beta.3` and was still observing the issue.
